televisionbingecalculator/src/main/java/com/vanillax/televisionbingecalculator/app/kotlin/viewmodels/DetailsViewModel.kt

```diff
@@ -137,6 +139,9 @@ class DetailsViewModel(theMovieDBService: TheMovieDBService, justWatchAPIService
     }
 
     fun selectSeason(seasonNumber: Int) {
+
+        //TODO is this right?
+        val detailsItemViewModel = detailsItemViewModel.value
```

Comment:
you can access the value from _detailsItemViewModel and use ?.let on it so that you don't need to do checks at use site.

_detailsItemViewModel.value?.let{
episodeCount = it.episodeCount.toString()
}

Reply:
fixed